core/src/actions/cluster.ts

```diff
@@ -128,11 +104,34 @@ export class ClusterClearCache extends AuthenticatedAction {
   }
 
   async runWithinTransaction() {
-    await cache.clear();
+    await clearRedis();
     return { success: true };
   }
 }
 
+async function clearRedis() {
+  await cache.clear(); // clear redis cache
+
+  // resque
+  await deleteKeys("*resque:queue:*"); // clear resque queues
+  await deleteKeys("*resque:delayed:*"); // clear resque delayed queues
+  await deleteKeys("*resque:timestamps:*"); // clear resque timestamps
+  await deleteKeys("*resque:*lock*:*"); // clear resque locks
+  await deleteKeys("*resque:stat:*"); // clear resque stats
+
+  // wait for any currently-running workers
+  await new Promise((resolve) => setTimeout(resolve, config.tasks.timeout * 2));
```

Comment:
This and the `keys` call can take a while. might need a UI accommodation of some sort.

Reply:
For me with 1K profiles, it's about 5s (most of the time being the sleep you've highlighted).  The UI buttons  have a loading state at least... 

![reset](https://user-images.githubusercontent.com/303226/109864534-cd933200-7c17-11eb-9515-55746e1806e5.gif)